ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java

```diff
@@ -307,4 +359,134 @@ private void updateContainersWithMySqlStore(short accountId, Collection<Containe
       }
     }
   }
+
+  public static class MySqlBackupFileManager {
+    private static final Logger logger = LoggerFactory.getLogger(BackupFileManager.class);
+    private static final String BACKUP_FILE_NAME = "MySqlAccountMetadataCopy";
+    private final AccountServiceMetrics accountServiceMetrics;
+    static final String TEMP_FILE_SUFFIX = "tmp";
+    static final String SEP = ".";
+    static final DateTimeFormatter TIMESTAMP_FORMATTER = DateTimeFormatter.ofPattern("yyyyMMdd'T'HHmmss");
+    static final ZoneOffset zoneOffset = ZoneId.systemDefault().getRules().getOffset(LocalDateTime.now());
+    static final Pattern filenamePattern = Pattern.compile(BACKUP_FILE_NAME + "\\.(\\d{8}T\\d{6})$");
```

Comment:
you might want to also add a begin anchor of "^" for your pattern. Something like
`"^" + BACKUP_FILE_NAME + "......"`